app/controllers/viewing_parties_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ViewingPartiesController < ApplicationController
+  before_action :authorize_user
+
+  def new
+    @movie = TMDBInteraction.movie_by_id(params[:movie_id])
+    @party = Party.new
+    @user = current_user
+  end
+
+  def create
+    @movie = TMDBInteraction.movie_by_id(params[:party][:movie_id])
+
+    begin
+      @party = Party.create!(party_data)
+
+      params[:party][:guests].keys.each do |guest_id|
+        @party.guests << User.find(guest_id)
+      end
+    rescue ActiveRecord::RecordInvalid
+      flash.now[:error] = 'Please fill out all fields with proper information'
+      @user = current_user
+      render :new
+    end
```

Comment:
I'm not familiar with this syntax. Can we go over it sometime today during our check in?